.github/workflows/release.yml

```diff
@@ -6,36 +6,67 @@ on:
 
 jobs:
   # Publish to GitHub followed by pypi
-  publish_python:
-    if: "${{ ! github.event.release.draft }}"
-    name: "Publish Python Packages"
+  build-package:
+    name: "Build package with poetry"
     runs-on: "ubuntu-24.04"
+    if: "${{ ! github.event.release.draft }}"
```

Comment:
On apps we also ensure it's running on a tag, as a double safeguard against untagged releases. Not sure if you have a use-case for it in core...

`if: "startsWith(github.ref, 'refs/tags/v')"`